### PR TITLE
Dashboard rework

### PR DESCRIPTION
**Pull request by davipatury**
Closes #57

<details>
<summary>Desktop Preview</summary>

</details>
<details>
<summary>Mobile Preview</summary>

</details>

###### [Back-end #983](https://github.com/SwitchbladeBot/switchblade/pull/983)

### CODE REVIEW

File: src/pages/TestDashboard.vue [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        configurable: true,
+        toggleable: true,
+        toggled: true
+      }, {
+        key: 'language',
+        displayName: 'Language',
+        description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
+        configurable: true
+      }]
+    }
```
Comment (davipatury) [inline src/pages/TestDashboard.vue]: Bot API response examples ^

File: src/pages/TestDashboard.vue [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+      feature.toggled = value
+      console.log(`"${feature.displayName}" is ${value ? 'on' : 'off'}`)
+    },
+    configurationModal (key) {
+      const feature = this.features && this.features.find(f => f.key === key)
+      if (!feature) return
+
+      console.log(`Open "${feature.displayName}" modal`)
+    }
```
Comment (davipatury) [inline src/pages/TestDashboard.vue]: TODO

**Comment by metehus** [comment]
We could make a message configuration, like for the joinlock, where the user can change the content and also create embeds with visualizers

**Comment by pedrofracassi** [comment]
I agree with embed visualizers. There are some great open source ones around.

As to where would we put that, it'd be inside the card's "advanced" modal that opens when you click the cog

**Comment by metehus** [comment]
i was thinking about the message configuration and thinking of variables like for a custom join message could use `%user%` or `%membercount%`

also i liked the idea of the fields been provided from the api but we need to put like
```
{
  "fields": [
    {
      "type": "string",
      "maxLength": 20,
      "minLength": 2
    }
  ]
}
```
wouldnt this be so much?

**Review by kaikecarlos — APPROVED** [review]

**Review by perronosaurio — APPROVED** [review]

**Review by pedrofracassi — APPROVED** [review]
In Davi we trust

**Pull request merged**